R/utilities.R

```diff
@@ -1,225 +1,260 @@
-#' Attach score result `score_res` to score object `score_obj` containig metadata
+#' Attach score result `results` to score object `score_obj` containig metadata
 #'
 #' @param x NULL
-#'
+#' @param results NULL
 #' @param ... NULL
 #'
 #' @export
-attach_score <- function(x, ...) {
-  UseMethod("attach_score")
-}
+attach_score <- S7::new_generic(
+  "attach_score",
+  "x",
```

Comment:
With the exception of the first argument of a function, I would err on the side of being verbose by using the argument names. 

```suggestion
  dispatch_args = "x",
```